lib/packages.js

```diff
@@ -0,0 +1,223 @@
+			}
+			return callback(null, result);
+		});
+	});
+};
+
+exports.find = function(name, callback)
+{
+	openCollection(function(error, collection)
+	{
+		if (error)
+		{
+			log.error('Could not open collection: %s', error);
+			return callback('database not available');
+		}
+		collection.findOne({name: name}, function(error, result)
+		{
+			if (error)
+			{
+				log.error('Could not read package %s: %s', name, error);
+				return callback('database not available');
+			}
+			return callback(null, result);
+		});
+	});
+};
+
+function testFind(callback)
+{
+	var name = 'test_package';
+	var value = '187349';
+	var object = {
+		name: name,
+		value: value,
+	};
+	exports.update(object, function(error)
+	{
+		testing.check(error, 'Could not save object', callback);
+		exports.find(name, function(error, result)
+		{
+			testing.check(error, 'Could not find package', callback);
+			testing.assert(result, 'Did not find package', callback);
+			testing.assertEquals(result.value, value, 'Invalid value', callback);
+			testing.success(callback);
+		});
+	});
+}
+
+exports.listAll = function(callback)
+{
+	openCollection(function(error, collection)
+	{
+		if (error)
+		{
+			log.error('Could not open collection: %s', error);
+			return callback('database not available');
+		}
+		collection.find({}, {name: true}).toArray(function(error, result)
+		{
+			if (error)
+			{
+				log.error('Could not list packages: %s', error);
+				return callback('database not available');
+			}
+			return callback(null, result);
+		});
+	});
+};
+
+function testListAll(callback)
+{
+	exports.listAll(function(error, result)
+	{
+		testing.check(error, 'Could not list packages', callback);
+		testing.assert(result, 'Did not list packages', callback);
+		testing.success(callback);
+	});
+}
+
+exports.remove = function(name, callback)
+{
+	openCollection(function(error, collection)
+	{
+		if (error)
+		{
+			log.error('Could not open collection: %s', error);
+			return callback('database not available');
+		}
+		collection.remove({name: name}, {w: 1}, callback);
+	});
+};
+
+function testRemove(callback)
+{
+	var name = 'to_remove';
+	var object = {
+		name: name,
+	};
+	exports.update(object, function(error)
+	{
+		testing.check(error, 'Could not save object', callback);
+		exports.find(name, function(error, result)
+		{
+			testing.check(error, 'Could not find package', callback);
+			testing.assert(result, 'Did not find package', callback);
+			exports.remove(name, function(error)
+			{
+				testing.check(error, 'Could not remove package', callback);
+				exports.find(name, function(error, result)
+				{
+					testing.check(error, 'Could not find package again', callback);
+					testing.assert(!result, 'Should not find package', callback);
+					testing.success(callback);
+				});
+			});
+		});
+	});
+}
+
+exports.close = function(callback)
+{
+	db.close(callback);
+};
+
+function testClose(callback)
+{
+	console.log('closing');
```

Comment:
I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.